### PR TITLE
Iso noise

### LINKED ISSUE #2149: 'photon_noise' cannot be set to 'ISO0' despite it being the default.
**Describe the bug**
According to the built in help in cjxl (`-v -h`), the `--photon_noise` parameter works as follows:
> --photon_noise=ISO3200
    Adds noise to the image emulating photographic film noise. The higher the given number, the grainier the image will be. As an example, a value of 100 gives low noise whereas a value of 3200 gives a lot of noise. The default value is 0.

However, if this is set to the 'default' of `0`, the following error appears:
> Error parsing flag --photon_noise=ISO0

**To Reproduce**
Run `cjxl` like so: `cjxl foo.png foo.jxl --photon_noise=ISO0`

**Expected behavior**
Ideally, the image would still be encoded, just with no photon noise at all, even though 'ISO0' doesn't make a whole lot of sense naming-wise.

**Environment**
 - OS: Linux
 - Compiler version: `clang version 14.0.6`
 - CPU type: x86_64
 - cjxl/djxl version string: `cjxl v0.8.1 [AVX2,SSE4,SSSE3,Unknown]`

### PR DESCRIPTION
**Pull request by mo271**
fixes #2149 

Also moving the "ISO" from the argument to the name of the flag.

### CODE REVIEW

File: tools/cjxl_main.cc [hunk 1]
```diff
@@ -54,7 +54,8 @@ namespace tools {
 
 namespace {
 inline bool ParsePhotonNoiseParameter(const char* arg, float* out) {
-  return strncmp(arg, "ISO", 3) == 0 && ParseFloat(arg + 3, out) && *out > 0;
+  return ParseFloat(arg, out) && *out >= 0;
+  ;
```
Comment (szabadka) [inline tools/cjxl_main.cc:58]: remove?

**Review by szabadka — APPROVED** [review]

File: tools/cjxl_main.cc (hunk 1 above)
Comment (mo271) [inline tools/cjxl_main.cc:58]:
```suggestion
  
```

File: tools/cjxl_main.cc [hunk 2]
```diff
@@ -54,7 +54,8 @@ namespace tools {
 
 namespace {
 inline bool ParsePhotonNoiseParameter(const char* arg, float* out) {
-  return strncmp(arg, "ISO", 3) == 0 && ParseFloat(arg + 3, out) && *out > 0;
+  return ParseFloat(arg, out) && *out >= 0;
+
```
Comment (mo271) [inline tools/cjxl_main.cc:58]:
```suggestion
  return ParseFloat(arg, out) && *out >= 0;
```

File: tools/cjxl_main.cc (hunk 1 above)
Comment (mo271) [reply tools/cjxl_main.cc]: done

**Review by szabadka — APPROVED** [review]

**Pull request merged**